packages/mgt-components/src/components/mgt-tasks/mgt-tasks.ts

```diff
@@ -1178,142 +1159,151 @@ export class MgtTasks extends MgtTemplatedComponent {
-        <div
-          class="TaskContent"
-          @click=${() => {
-            this.handleTaskClick(task);
-          }}
-        >
-          <span
-            class=${classMap({
-              Complete: completed,
-              Incomplete: !completed,
-              TaskCheckContainer: true
-            })}
-            @click="${(e: UIEvent) => {
-              if (!this.readOnly) {
-                if (!task.completed) {
-                  void this.completeTask(task);
-                } else {
-                  void this.uncompleteTask(task);
-                }
-
-                e.stopPropagation();
-                e.preventDefault();
-              }
-            }}"
-            @keydown="${(e: KeyboardEvent) => {
-              if (e.code === 'Enter') {
-                if (!this.readOnly) {
-                  if (!task.completed) {
-                    void this.completeTask(task);
-                  } else {
-                    void this.uncompleteTask(task);
-                  }
-
-                  e.stopPropagation();
-                  e.preventDefault();
-                }
-              }
-            }}"
-          >
-            ${taskCheck}
-          </span>
-          <div class="TaskDetailsContainer ${this.mediaQuery} ${this._currentGroup ? 'NoPlan' : ''}">
-            ${taskDetails}
+        id="task-${task.id}"
+        class=${taskClasses}
+        @click=${() => this.handleTaskClick(task)}>
+        <div class="TaskDetailsContainer">
+          <div class="Top">
+            <div class="CheckAndTitle">
+              <fluent-checkbox
+                @click=${(e: MouseEvent) => this.checkTask(e, task)}
+                @keydown=${(e: KeyboardEvent) => this.handleTaskCheckKeyDown(e, task)}
+                ?checked=${completed}>
+                  ${name}
+              </fluent-checkbox>
+            </div>
+            <div class="TaskOptions">${taskOptions}</div>
           </div>
-          ${taskOptions}
-
+          <div class="Bottom">${taskDetails}</div>
         </div>
       </div>
     `;
   }
 
-  private renderAssignedPeople(task: ITask) {
-    let assignedPeopleHTML = null;
-    let assignedGroupId: string;
+  private async handleTaskCheckKeyDown(e: KeyboardEvent, task: ITask) {
+    if (e.key === 'Enter') {
+      if (!this.readOnly) {
+        if (!task.completed) {
+          await this.completeTask(task);
+        } else {
+          await this.uncompleteTask(task);
+        }
+
+        e.stopPropagation();
+        e.preventDefault();
+      }
+    }
+  }
+
+  private async checkTask(e: MouseEvent, task: ITask) {
+    if (!this.readOnly) {
+      const target = this.shadowRoot.querySelector(`#task-${task.id}`);
+      if (target) target.classList.add('Updating');
+      if (!task.completed) {
+        await this.completeTask(task);
+      } else {
+        await this.uncompleteTask(task);
+      }
+      // wait a bit before removing the freeze on the task updating
+      debounce(() => {
```

Comment:
Why are we adding an artificial UI lag here? We're already awaiting the server call, we should update the UI back to the 'natural' state as soon as possible after the server side change is completed IMO

Reply:
Oh yes! I am removing it.